Add initial project files

ygoumaz opened the pull request:
ditto.

ygoumaz committed:
add files

ygoumaz committed:
Update README.md

ygoumaz committed:
Fix readme

ygoumaz committed:
Add readme note

ygoumaz committed:
fix readme

BenjaminDeleze commented on Dockerfile:
```diff
@@ -0,0 +1,6 @@
+FROM cyberbotics/webots:R2022b-ubuntu20.04
```
```suggestion
FROM cyberbotics/webots.cloud:R2022b
```

BenjaminDeleze commented on README.md:
````diff
@@ -1,4 +1,34 @@
+COPY . $PROJECT_PATH
+```
+
+The following two additional lines allow to compile the controller and the plugin when [webots.cloud](webots.cloud) creates the Docker image.
+
+```dockerfile
+RUN cd $PROJECT_PATH/controllers/e-puck && make clean && make
+RUN cd $PROJECT_PATH/plugins/remote_controls && make clean && make
+```
````
I think you also need to compile the robot window.

I don't know if it make a lot of sense to put the remote_controls plugin as the user will not be able to test it in most of the case

BenjaminDeleze reviewed (changes requested):
Seems good, some suggestions:

ygoumaz committed:
Fix base docker image

ThomasOliverKimble reviewed (changes requested):
Same comments as Benjamin. Minor detail: I would suggest to open the world in Webots and save it again to create a thumbnail. On save it should be a hidden file that would be uploaded to github

ygoumaz committed:
Cleanup

ygoumaz committed:
Update README.md

ygoumaz committed:
Update urls in proto epuck

ygoumaz committed:
Remove remote controls

ygoumaz committed:
Cleanup

ygoumaz committed:
Change viewpoint + screenshot

ygoumaz commented:
I have cleaned the repository and added your suggestions. I have tried to start the world on webots.cloud but unfortunately I still get the same error message telling me that it can't find the .wbt file. Can it be related to its name containing a hyphen symbol ('-') ?

BenjaminDeleze commented:
It could be (even so it should not), could you try to rename the world to test?

ygoumaz committed:
Renamed world

ygoumaz committed:
renamed all

BenjaminDeleze reviewed (commented):
Seems good to me, but I think we should understand and fix the problem before merging

ygoumaz commented:
Sure we should fix the problem, changing the name to e_puck.wbt still triggers the error message.

ThomasOliverKimble commented:
>  Can it be related to its name containing a hyphen symbol ('-') ?

This used to be a problem but it was fixed in `simulation.php` here #[3](https://github.com/cyberbotics/webots-cloud/pull/3)

ygoumaz committed:
New world

ygoumaz commented:
@ThomasOliverKimble Could you check why it fails? The only reason I see is a wrong parsing of the URL, as the same project works in your own repo: https://github.com/ThomasOliverKimble/GuidedTour/blob/guided-tour/e-puck/worlds/e-puck.wbt

ygoumaz committed:
Update Dockerfile

ygoumaz committed:
Update texture urls

ygoumaz committed:
Cleanup and rename world files

ygoumaz commented:
The issue with the world loading has been fixed in [#4874](https://github.com/cyberbotics/webots/pull/4874). The repository is now completely cleaned and works fine on webots.cloud.

BenjaminDeleze reviewed (dismissed):
Thank you!

ygoumaz committed:
Add Python controller

ygoumaz committed:
update controller

ygoumaz committed:
Remove c controller compilation

ygoumaz committed:
Update e-puck.py

ygoumaz committed:
Fix epuck.py

ThomasOliverKimble commented:
> Sure we should fix the problem, changing the name to e_puck.wbt still triggers the error message.

fixed with webots PR #[4874](https://github.com/cyberbotics/webots/pull/4874)

ygoumaz committed:
Remove old c controller

ygoumaz committed:
epuck.py cleanup

ygoumaz committed:
Update README.md

ygoumaz commented:
I have replaced the C controller by a Python one. It should be equivalent.

ygoumaz committed:
Update e-puck.py

ygoumaz committed:
Merge branch 'feature-add-initial-files' of https://github.com/cyberbotics/webots-cloud-simulation-template into feature-add-initial-files

BenjaminDeleze reviewed (approved):
Thank you!

ThomasOliverKimble reviewed (approved):
Thank you!
